pkg/asset/installconfig/openstack/validation/machinepool.go

```diff
@@ -57,25 +58,17 @@ func ValidateMachinePool(p *openstack.MachinePool, ci *CloudInfo, controlPlane b
 	return allErrs
 }
 
-func validateZones(input []string, actual []string, fldPath *field.Path) field.ErrorList {
-	allErrs := field.ErrorList{}
-
-	// Base Case: No user input
-	if len(input) == 1 {
-		if input[0] == "" {
-			return allErrs
-		}
+func validateZones(input []string, available []string, fldPath *field.Path) field.ErrorList {
+	// check if machinepool default
+	if len(input) == 1 && input[0] == "" {
+		return field.ErrorList{}
```

Comment:
does the default machine pool in install-config still have "" element in zones?

Reply:
Yes, an earlier version of this patch attempted to change it to empty list, but the teams preference was to set the default like this: https://github.com/openshift/installer/pull/4009#discussion_r465708566